src/event/writer.rs

```diff
@@ -118,9 +123,18 @@ impl EventWriter {
     pub async fn write_event(&mut self, event: Vec<u8>) -> oneshot::Receiver<Result<(), Error>> {
         let size = event.len();
         let (tx, rx) = oneshot::channel();
+        let (tx_flush, rx_flush) = oneshot::channel();
         let routing_info = RoutingInfo::RoutingKey(None);
+        let routing_info_flush = RoutingInfo::RoutingKey(None);
         if let Some(pending_event) = PendingEvent::with_header(routing_info, event, None, tx) {
             let append_event = Incoming::AppendEvent(pending_event);
+            if let Some(pending_event_flush) =
```

Comment:
Should'nt we do this logic post writring append_event? As I understand with this logic `flush()` would still have one event outstanding, right?

Reply:
I think youre right! Let me fix it.